sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -217,11 +218,12 @@ def delete_batch(self, paths):
     """
     final_results = []
     s = 0
-    while s < len(paths):
-      if (s + MAX_BATCH_OPERATION_SIZE) < len(paths):
-        current_paths = paths[s:s + MAX_BATCH_OPERATION_SIZE]
+    path_list = list(iter(paths))
```

Comment:
minor: shall we just convert when its not a list:
```
if not isinstance(paths, list): paths = list(iter(paths))
```

Reply:
done.